packages/ui-causal-graph-editor/src/shared/use-causal-graph-editor.tsx

```diff
@@ -52,13 +53,16 @@ export default function useCausalGraphEditor(
     const [state, dispatch] = useReducer(
         GraphReducer,
         {
-            editorMode,
             newNodesRequirePosition,
         },
         (initState: GraphState) => {
+            const parsedGraph = causalGraphParser(graphData, availableInputs);
+            const newEditorMode = editorMode ?? (isDag(parsedGraph) ? EditorMode.DEFAULT : EditorMode.PAG_VIEWER);
```

Comment:
Looks good - quick question, should the effect below (which invokes INIT_GRAPH) also have this logic? Since the graph can change from the outside and perhaps it becomes a non-DAG

Reply:
We can but that can cause the editor mode to switch on users so as they add edit edges as the panels shown would change which seems confusing behaviour

Reply:
In what cases is the editor mode undefined here in the first place? 

I'm not sure what the intuitive behaviour is but let's make sure the graph viewer can't break if we first assume it's a DAG and then a cycle is added - we prevent it via the UI but it can change from the outside, what happens then?

Reply:
If the user doesn't define it, it would be undefined here. Unless they set the layout to Planar in that case we define the editorMode for them.

If we then change graphData from the outside to include a cycle. If then the user tries to add further edges, the current behaviour is:
- If the newly added edge connects to the cycle then they can't add it as on the onAddEdge handler we check that the added edge doesn't create a cycle and since a cycle is already present it seems to get confused and not let further edges to be added. 
- If the edge is added in such a way that it does not interact directly with the cycle then the component breaks. Note that this is a way in which we can currently break the component, it is something that could happen without considering any of the changes done in this ticket. 
- When looking into this now, I also found that by accepting edges one can create a cycle in resolver mode without any errors being raised until further edges are added as explained above.

Reply:
Yeah, seems like we need to properly handle cycles being created by changes made from the outside. Perhaps that's a separate ticket but we'll definitely need to look into it to make sure you can't just 'break' the component and at least display meaningful errors. Can you add a ticket for it?

Reply:
Added. For this ticket what should I do? A graph is a dag if it does not have a cycle and all edges are directed, so don't think it makes that much sense to switch between editor modes given the second condition even if graphdata is changed, what do you think?

Reply:
Yeah maybe it makes sense to run the same inference logic both in the initializer and in the effect if they don't explicitly specify the mode